### ISSUE TITLE
Enhancement Request: Easy selection of multiple messages on a conversation and multiple conversations

### ISSUE THREAD

**Issue by fcastilloec** [qklabs/qksms-tracker#42]
There's no way to select multiple messages under a conversation to either copy, export or delete them (or any other action that I can't think of). The same is true for conversations.

I've seen other apps handle this by either adding a option of "Select" under the long press menu, or the 3-dot menu. This option selects the message (or conversation) and lets you click on others to keep selecting. The 3-dot menu would have all the actions you can perform to those messages (or conversations).

**Comment by moezbhatti** [comment]
You're completely right, it would be an awesome feature to have. This will be one of the next few things I tackle

**Comment by moezbhatti** [comment]
Renamed due to current state of issue

**Comment by Gitoffthelawn** [comment]
+1

**Comment by themelisx** [comment]
It will be great to multiselect and delete messages in a conversation

**Comment by realmain** [comment]
Since #874 was implemented, should this one be too before the next release?

**Comment by realmain** [comment]
This is how SilenceIM does it that I kinda like:

![](https://i.imgur.com/aZOGATx.png)

![](https://i.imgur.com/Ucz8kaw.png)

Copy still works for multi-select. Each message is copied with a newline/line break, from top to bottom. Like in the case of the last pic, it'd be:

```
Bsgs
Sbba
Ss
```
I rarely use it, but it's useful sometimes.

**Comment by amyers3** [comment]
I prefer the Gmail multi selection, with check marks over the Avatar or like a Textra where check boxes come out from the side. I think this is a key feature though
